### PR TITLE
basic workflow example draft

### PR DESCRIPTION
**Pull request by laerling**
This PR is for discussing and writing the basic workflow example together, since it is crucial that the example is written with full consent of both of us. Is this PR a valid way of doing this, or would an issue be better?
  
I've put the basic workflow example into the design document. Do you think it should be an extra document instead?
  
It will probably be the best to discuss and complete the example when we meet again next time.

### CODE REVIEW

**Comment by majewsky** [comment]
> Do you think it should be an extra document instead?

Yes.

> It will probably be the best to discuss and complete the example when we meet again next time.

Yes.

File: spec/design.md [hunk 1]
```diff
@@ -42,3 +42,12 @@ The following principles guide and inform the design of the VT6 protocols and th
...
 5. **Ease of use** - VT6 commands should be human-readable, and the protocol should be as easy to emit and consume in all programming and scripting languages as long as the ease of use does not significantly impede performance or resource requirements.
+
+## 3. Basic workflow example
+
+This example describes the example of a usual workflow when using a VT6-capable terminal.
+This terminal acts as a VT6 server.
+When it starts up it typically calls a VT6 shell which acts as a VT6 client.
+From this shell several more VT6 clients and legacy clients might be started.
+All clients talk to the original VT6 server, in this case the terminal.
+In order for legacy clients to be able to communicate with the VT6 server they are each wrapped by a VT6-capable wrapper which acts as a VT6 client.
```
Comment (majewsky) [inline spec/design.md]: Not necessarily. Something like `cat` or `grep` does not need to care about VT6 at all. They just run as normal, they just use the stdin/stdout supplied by the shell.

**Comment by laerling** [comment]
>> Do you think it should be an extra document instead?

> Yes.

Actually, I don't see why it should be an extra document right now. It is a general explanation of the basic workflow of vt6 and therefore is part of the design. This and further 'examples', which will contain special cases of the vt6 workflow, explain the design of the backend of vt6.
Please explain why it should be an extra document.

**Comment by majewsky** [comment]
@hkgit03 and I split `spec/design.md` into `spec/why.md` and `spec/how.md`. This branch inserts the first content into `spec/how.md`.

**Comment by majewsky** [comment]
Additional bullet points that were on the whiteboard during our meeting on Friday:

- As for describing the features of a terminal, "VT6-compatible" is completely orthogonal to "VT100/Xterm/etc.-compatible". A VT6 terminal can (and probably should) support some set of ANSI escape codes as well (and advertise them via `setenv("TERM", ...)`).
- A legacy client therefore does not need wrapping, unless the terminal is not legacy-compatible.

**Comment by laerling** [comment]
@majewsky I added exhaustive examples.

Questions:
- Should we call a "wrapper" a "proxy" instead (and use "proxied" instead of "wrapped")?
- 1.1. "(...) and a vt6 message channel". Is there only one socket which is used by all clients? This is not clear to me as of now.
  If so, we should instead write "(...) and **the** vt6 message channel" (without emphasis, of course).

TODO:
- Insert cross-references to the core spec

**Comment by laerling** [comment]
Also TODO: Include @majewsky's notes about orthogonality and wrapping.

**Comment by majewsky** [comment]
img_20171107_214422

- include usecase from above image (demuxing)
- split the large image in section 3 into separate ones

**Comment by laerling** [comment]
I consider `how.md` to be finished, this branch therefore ready to be merged. What do you think?

**Comment by majewsky** [comment]
What we have in here right now appears to be correct (as in: truthful to our vision), though probably subject to revision once we figure out how multiplexing is actually going to work in practice.

**Pull request merged**